tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh

```diff
@@ -103,8 +106,45 @@ EOF
 	exit "${return_code}"
 }
 
+cleanup_and_fail() {
+	rm -f "${component_tarball_path}"
+	return 1
```

Comment:
More of a nit than anything, but do we know if this will always fail due to an `Operation not permitted`?

Reply:
So, it seems there's something fishy with my changes (exposed by the CI, as I'm not able to reproduce this locally).

In some part of the code, which I still need to find out where, we're not adding the `git config --global --add safe.directory` to the Kata Containers repo, resulting on the errors raised by the CI.

With that out of the way, the `cleanup_and_fail()` function is not related to that.  What it does, though, is ensure that we return some sort of fail in case we cannot use cached components, and then we proceed with the normal build of those, as either the component itself is not cached, or the cache is not up-to-date, or the image used to build the component changed.

Reply:
Okay, I found the issue aboute the missing `git config --global --add safe.directory ...`

The error should not be present in the tests anymore.

Reply:
Gotcha. There was that, but also that error code return `1` is specifically signifying an error which is `Operation not permitted.` Maybe I can create a follow-up PR to add some error passing here to help sift through the possible variety of the errors?